app/nodejs/app.js

```diff
@@ -19,26 +19,71 @@ app.js - application
 'use strict';
 
 import express from 'express';
-import logger from './logger.js'
-import { engine } from 'express-handlebars';
+import logger from './logger.js';
+import {engine} from 'express-handlebars';
+import {Storage} from '@google-cloud/storage';
 
 const app = express();
+const storage = new Storage();
 
 app.engine('handlebars', engine());
 app.set('view engine', 'handlebars');
 app.set('views', './views');
 
+/**
+ * Retrieve data from Cloud Storage, based on querystrings
+ * @param {string} fur Fur type (Gray, Black, Cinnamon)
+ * @param {string} age Age (Adult, Juvenile)
+ * @param {string} location Location (Above Ground, Ground Level)
+ */
+async function retrieveData(fur, age, location) {
+  const dataFilename = [fur, age, location].join('/') + '/data.json';
+
+  const dataFile = storage
+      .bucket(process.env.PROCESSED_DATA_BUCKET)
+      .file(dataFilename);
+
+  if (!dataFile.exists()) {
```

Comment:
This isn't returning synchronously, and a promise is truthy, so the logic here doesn't fire if the file doesn't exist.

Reply:
Fixed it (hopefully!)